core/src/test/java/org/elasticsearch/index/query/BaseQueryTestCase.java

```diff
@@ -396,21 +401,56 @@ protected static void assertValidate(QueryBuilder queryBuilder, int totalExpecte
 
     /**
      * create a random value for either {@link BaseQueryTestCase#BOOLEAN_FIELD_NAME}, {@link BaseQueryTestCase#INT_FIELD_NAME},
-     * {@link BaseQueryTestCase#DOUBLE_FIELD_NAME} or {@link BaseQueryTestCase#STRING_FIELD_NAME}, or a String value by default
+     * {@link BaseQueryTestCase#DOUBLE_FIELD_NAME}, {@link BaseQueryTestCase#STRING_FIELD_NAME} or
+     * {@link BaseQueryTestCase#DATE_FIELD_NAME}, or a String value by default
      */
-    protected static Object randomValueForField(String fieldName) {
-        Object value;
+    protected static Object getRandomValueForFieldName(String fieldName) {
+        Object value = randomAsciiOfLengthBetween(1, 10);
         switch (fieldName) {
-            case BOOLEAN_FIELD_NAME: value = randomBoolean(); break;
-            case INT_FIELD_NAME: value = randomInt(); break;
-            case DOUBLE_FIELD_NAME: value = randomDouble(); break;
-            case STRING_FIELD_NAME: value = randomAsciiOfLengthBetween(1, 10); break;
-            default : value = randomAsciiOfLengthBetween(1, 10);
-        }
+            case STRING_FIELD_NAME:
+                value = rarely() ? randomUnicodeOfLength(10) : value; // unicode in 10% cases
+                break;
+            case INT_FIELD_NAME:
+                value = randomIntBetween(0, 10);
+                break;
+            case DOUBLE_FIELD_NAME:
+                value = randomDouble() * 10;
+                break;
+            case BOOLEAN_FIELD_NAME:
+                value = randomBoolean();
+                break;
+            case DATE_FIELD_NAME:
+                value = new DateTime(System.currentTimeMillis(), DateTimeZone.UTC).toString();
+                break;
+        } // all other fields assigned to random string
         return value;
     }
 
     /**
+     * Helper method to return a mapped or a random field
+     */
+    protected String getRandomFieldName(String... exclude) {
+        // if no type is set then return a random field name
+        if (currentTypes == null || currentTypes.length == 0 || randomBoolean()) {
+            return randomAsciiOfLengthBetween(1, 10);
+        }
+        // handle exclude fields
+        List<String> mappedFieldNames = Lists.newArrayList(BaseQueryTestCase.mappedFieldNames);
+        for (String fieldName : exclude) {
+            mappedFieldNames.remove(fieldName);
+        }
```

Comment:
maybe we can simply keep a static copy of the mappedFieldNames that doesn't contain OBJECT_NAME rather than having to remove it all the time?